File: sandbox/testing_consumer.py
from __future__ import absolute_import

from consumer import Consumer


def callback_ftn(arg):
    print(arg)


config = {'userName': 'guest',
          'password': 'guest',
          'host': 'localhost',
          'port': 5672,
          'virtualHost': 'rvhost',
          'queueName': 'default',
          'exchangeName': 'exchange',
          'routingKey': 'routingkey',
          'exchangeType': 'direct',
          'exchangeOptions': {'passive': False,
                              'durable': False,
                              'autoDelete': True,
                              'internal': False},
          'queueOptions': {'passive': False,
                           'durable': False,
                           'exclusive': False,
                           'autoDelete': True}
          }

print(config)
if __name__ == '__main__':

    with Consumer(config) as csmr:

        csmr.consume(message_received_callback=callback_ftn)

File: nodejs_app/testing-consumer.js
var RabbitMQConsumer = require('./server/rabbitmq-consumer'); //specify the name of the file here containing the code we just wrote

var configuration = {
    userName: "user",
    password: "abc123",
    host: "rabbitmq.mydomain.com",
    port: 5671,
    virtualHost: "home",
    exchangeName: "home_exchange",
    exchangeType: "topic",
    exchangeOptions: {
        durable: true,
        internal: false,
        autoDelete: false
    },
    queueName: "roku_remote_queue",
    queueOptions: {
        exclusive: false,
        durable: true,
        autoDelete: false
    },
    routingKey: "home.remote.roku"
};

var consumer = new RabbitMQConsumer(configuration);

consumer.on('received', function(messageText) {
    console.log('Message received:', messageText);
});

consumer.consume();